nodes/validator/main.py

```diff
@@ -0,0 +1,83 @@
+import json
+import hashlib
+import aiohttp
+import asyncio
+from aiohttp import web
+import os 
+import random
+
+async def transaction_new(request):
+    try:
+        txn = await request.json()
+        if await is_valid(txn):
+            pass
+        return web.Response(status = 200)
+    except:
+        return web.Response(status = 500)
+
+
+async def is_valid(txn: dict = {}) -> bool:
+    hash = txn['hash']
+    # Create a new dictionary without 'hash' and 'signature' keys
+    hashables = {}
+    for key, val in txn.items():
+        if not key in ["hash", "timestamp", "signature"]:
+            hashables[key] = val
+    # Dump string minified (i.e. no spaces around separators)
+    hashable_str = json.dumps(
+        hashables,
+        separators = (",",":")
+    )
+    # Encode string as bytestring
+    hashable_bytes = hashable_str.encode('utf-8')
+    # Calculate the SHA256 hash of the bytes
+    sha256_hash = hashlib.sha256()
+    sha256_hash.update(hashable_bytes)
+    checksum = sha256_hash.hexdigest()
+    # Return result of hex string comparison
+    return checksum == hash
+
+
+async def transact_handler(request):
+    try:
+        data = await request.json()
+        key = data.get('key')
+        txn = data.get('txn')
+        sig = data.get('txn', {}).get('signature')
```

Comment:
We can pass the whole `data` packet to the verifier; we don't need to separate keys anymore.